common/logger/src/struct_log.rs

```diff
@@ -215,6 +237,14 @@ impl StructuredLogEntry {
         self
     }
 
+    #[doc(hidden)] // set from macro
+    pub fn add_category(&mut self, target: &'static str) -> &mut Self {
+        if self.category.is_none() {
+            self.category = Some(target);
+        }
+        self
+    }
```

Comment:
This is slightly different behavior than the other ones.  Is there a reason for that it can't be overridden?

Reply:
I didn't let it overwrite because otherwise the macro would overwrite 100% of these to be just the module-name. So since I didn't have context on its use I just tried to be extra careful.